ocaml/xapi/xapi_gpumon.ml

```diff
@@ -46,4 +46,22 @@ module Nvidia = struct
     let get = Gpumon_client.Client.Nvidia.get_pgpu_metadata in
     [key, get dbg pgpu_pci_address]
 
+  let get_pgpu_vm_compatibility ~__context ~vm ~pgpu =
+    let open Gpumon_client in
+    let dbg = Context.string_of_task __context in
+    let vm_domid = Int64.to_int (Db.VM.get_domid ~__context ~self:vm) in
+    let cur_pgpu_address =
+      Db.VM.get_VGPUs ~__context ~self:vm
+      (* currently we support only 1 nvidia vgpu *)
```

Comment:
...and we're relying on this function never being called on a VM with no vGPU (or with a non-nVIDIA one).